src/smart.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * Provides conversions from smart contract XDR values ({@link xdr.ScVal}) to
+ * native JavaScript types.
+ *
+ * @example
+ * import { SmartParser, ScInt, xdr } from 'stellar-base';
+ *
+ * let gigaMap = {
+ *   bool: true,
+ *   void: null,
+ *   u32: xdr.ScVal.scvU32(1),
+ *   i32: xdr.ScVal.scvI32(1),
+ *   u64: 1n,
+ *   i64: -1n,
+ *   u128: new ScInt(1).toU128(),
+ *   i128: new ScInt(1).toI128(),
+ *   u256: new ScInt(1).toU256(),
+ *   i256: new ScInt(1).toI256(),
+ *   map: {
+ *     arbitrary: 1n,
+ *     nested: 'values',
+ *     etc: false
+ *   },
+ *   vec: ['same', 'type', 'list'],
+ * };
+ *
+ * // then, simply:
+ * let scv = SmartParser.toScVal(gigaMap);
+ * // so scv.switch() == xdr.ScValType.scvMap()
+ *
+ * // then...
+ * someContract.call("method", scv);
+ */
+import xdr from './xdr';
+
+import { Address } from './address';
+import { ScInt, scValToBigInt } from './numbers/index';
+
+/**
+ * A container class for methods that convert to<-->from raw {@link xdr.ScVal}s.
+ */
+export class SmartParser {
```

Comment:
Generally, I'd rather see this as `xdr.ScVal.fromJson(j: any)`, `scval.toJson()`. Having it be methods on the existing `ScVal` type would merge better with our existing patterns, and avoid introducing a new "tool" to learn. But this is useful as is, so I don't think it's worth delaying the the release to bikeshed that.

Reply:
I'd love to shove it into the `xdr.ScVal` definition, but this is non-trivial because of the `js-xdr` nonsense (i.e. `xdr.ScVal` isn't a JavaScript object and so you can't just `.prototype.fromJson = ...` it). So I want to look into this but agreed about bikeshedding.